app/utils/base.py

```diff
@@ -1,5 +1,125 @@
 import re
+from app.api.exceptions import HTTPResponseCustomized
+from fastapi import Depends
+from app import crud, dependencies
+from app import dependencies
+from fastapi.exceptions import RequestValidationError
+from app.schemas import ServiceMetricCreate, MetricCreate, scoreCard, microserviceScoreCard, ScorecardServiceMetric
+from fastapi import APIRouter, Depends, Request, exception_handlers, status, Response, HTTPException
+from fastapi.responses import JSONResponse, PlainTextResponse
+from fastapi.encoders import jsonable_encoder
+from pydantic import BaseModel, Field
+from app.crud import CRUDMetric, CRUDServiceMetric, CRUDMicroserviceScoreCard, CRUDMicroservice
+from app import schemas, models, crud
+from typing import Any, List
+import json
+from fastapi.routing import APIRoute
+
+db_session = dependencies.get_db()
+metricCrud: crud.CRUDMetric = CRUDMetric(db_session)
+
 
 def format_code(name):
-  code = re.sub(r'\s+', '-', name.strip())
-  return code
\ No newline at end of file
+    code = re.sub(r'\s+', '-', name.strip())
+    return code.lower()
+
+
+def check_unique_ids(objects):
+    id_set = set()
+    for obj in objects:
+        if obj.id in id_set:
+            raise HTTPResponseCustomized(
+                status_code=400, detail="Metrics IDs are not unique, Please don't duplicate your metric")
+        id_set.add(obj.id)
+    return True
+
+
+def check_metric_weight(objects):
+    sum = 0
+    for obj in objects:
+        sum += obj.weight
+    if (sum == 100):
+        return True
+    raise HTTPResponseCustomized(
+        status_code=400, detail="Sum of metric weight is not 100")
+
+
+def check_metric_type(objects):
+    for obj in objects:
+        print(obj.type)
+        print(obj.desiredValue)
+        if (obj.type == "integer"):
+            if (not isinstance(obj.desiredValue, int)):
+                raise HTTPResponseCustomized(
+                    status_code=400, detail="desierdValue is not correct for integer metric")
+        elif (obj.type == "boolean"):
+            if (not isinstance(obj.desiredValue, bool)):
+                raise HTTPResponseCustomized(
+                    status_code=400, detail="desierdValue is not correct for boolean metric")
+        elif (obj.type == "string"):
+            if (not isinstance(obj.desiredValue, str)):
+                raise HTTPResponseCustomized(
+                    status_code=400, detail="desierdValue is not correct for string metric")
+        elif (obj.type == "float"):
+            if (not isinstance(obj.desiredValue, float)):
+                raise HTTPResponseCustomized(
+                    status_code=400, detail="desierdValue is not correct for float metric")
+
+
+criteriaStates = ("greater", "smaller", "equal",
```

Comment:
this is already defined in the models right ? we should define const variables like this only once and use its definition without re-defining it everytime we use  it